recipes/lockscreen.md

```diff
@@ -0,0 +1,193 @@
+# Lockscreen
+
+_By [adamlubek](https://github.com/adamlubek)_
+
+This recipe demonstrates RxJs implementation of lockscreen functionality (known for example from smartphones).
+
+<div class="ua-ad"><a href="https://ultimatecourses.com/?ref=76683_kee7y7vk"><img src="https://ultimatecourses.com/assets/img/banners/uc-leader.svg" style="width:100%;max-width:100%"></a></div>
+
+### Example Code
+
+(
+[StackBlitz](https://stackblitz.com/edit/rxjs-lockscreen?file=index.ts&devtoolsheight=30)
+)
+
+#### index.ts
+```js
+/*
+  Use mouse to 'swipe' across the lock pad (hold mouse button and swipe :) ).
+  Pad will turn green if password is correct or red if password is incorrect.
+  You can set password to whatever sequence you like.
+*/
+
+// RxJS v6+
+import { from, fromEvent, Subject, merge } from 'rxjs';
+import { switchMap, takeUntil, repeat, tap, map, throttleTime, distinctUntilChanged, filter, toArray, sequenceEqual } from 'rxjs/operators';
+import { pads, resetPasswordPad, setResult, setTouched, updateCurrentPassword } from './dom-updater';
+
+const sub = new Subject();
+const expectedPasswordUpdate$ = fromEvent(document.getElementById('expectedPassword'), 'keyup')
+  .pipe(
+    map((e: any) => e.target.value),
+    tap(pass => sub.next(pass.split('').map(e => parseInt(e))))
+  );
+let expectedPassword = [1, 2, 5, 2];
+const expectedPassword$ = sub.pipe(tap((v: any) => expectedPassword = v));
+
+const checkIfPasswordMatch = password => from(password).pipe(sequenceEqual(from(expectedPassword)));
+
+const actualPassword$ = fromEvent(document, 'mousedown')
+  .pipe(
+    tap(resetPasswordPad),
+    switchMap(_ => fromEvent(document, 'mousemove')),
+    takeUntil(fromEvent(document, 'mouseup')),
+    throttleTime(50),
+    map(({ clientX, clientY }: MouseEvent) => ({ x: clientX, y: clientY })),
+    map(v => pads.find(r =>
+      v.x > r.left &&
+      v.x < r.right &&
+      v.y > r.top &&
+      v.y < r.bottom)),
+    map(v => v ? v.id : v),
```

Comment:
Can we just return `v.id` here, it should either be valid or `undefined`?

Reply:
the reason for this is that when you swipe over white spaces between the dots, they don't have id. However, thinking about that I switched the order, I filter undefined out and then map as per your advice in next comment. looks better now, thanks :)

Reply:
even better, I actually replaced map with pluck